packages/api/src/external/commonwell/document/shared.ts

```diff
@@ -39,3 +48,46 @@ export function getFileExtension(value: string | undefined): string {
   const fileExtension = extension(value);
   return fileExtension ? `.${fileExtension}` : "";
 }
+
+export function getExtraDocRefInfo(content: DocumentContent | undefined): DocumentReferenceExtras {
+  const extraInfo: DocumentReferenceExtras = {};
+  const contained = content?.contained;
+  const context = content?.context;
+
+  if (contained) {
+    contained.forEach(item => {
+      const containedName = item.name;
+      if (item.resourceType === "Organization") {
+        if (typeof containedName === "string") {
+          extraInfo.organization = {
+            name: containedName,
+          };
+        }
+      } else if (item.resourceType === "Practitioner" && containedName) {
+        if (Array.isArray(containedName)) {
+          const name = containedName[0];
+          if (name) addPractitionerName(extraInfo, name);
+        } else if (typeof containedName === "object") {
+          addPractitionerName(extraInfo, containedName);
+        }
+      }
```

Comment:
I know we want to improve the quality of data we're storing and making available to our cxs, but I think there's a big risk to be losing data w/ these. For example, we're not storing the Org if the name is not a string, and as we see on the ticket, it might come as an object. Even if we process the object, it might come as an array of objects. At the end, we might as well leave as is and improve later - based on information we have stored, instead of assumptions that will lead to not having info to assess b/c we discarding it.

For Practitioner its the same thing. Sure, it might be a practitioner that has the org's name, but better that than not storing the actual Practitioner b/c we filtered it out, IMO.

I know we'll want @Goncharo to chime in since he wrote the ticket, now he's pinged. 😄